pkg/reconciler/trigger/trigger.go

```diff
@@ -62,6 +62,8 @@ var (
 	// defaulting fails.
 	defaultBackoffDelay  = "PT1S"
 	defaultBackoffPolicy = eventingduckv1beta1.BackoffPolicyExponential
+	// clusterRegionGetter is a function that can get the cluster region
+	clusterRegionGetter = utils.NewClusterRegionGetter()
```

Comment:
I suggest injecting this into the reconciler. For this and the others. Have Wire generate it and push it into the controller creation.

Reply:
This function is not really configurable in production. I am using it this way simply to overwrite it during testing.

Reply:
Another concern I originally had to do it this way is to avoid failing the controller when it cannot get the cluster region, as it is not required if there are valid setting already. But I think if we don't have to worry about this, we can just have a `clusterRegion string` in the Reconciler and let the controller initiate it properly. WDYT?